### DISCUSSION TITLE
Warhammer API

### DISCUSSION THREAD

**Discussion by brittonhayes** [brittonhayes/aos#2] [General]
> Hello,
>
> Sorry for writing and issue in the wrong repo but I can't make any issues in the warhammer repo.
> I am creating my own age of sigmar mobile app and your api seems exactly what I need. I was wondering where I can get the api key for the warhammer api if it is needed. It seemed to me that it is needed to use the api.
>
> Thank you in advance!
>
> Friendly regards,
> Morteeke

@morteeke Thanks so much for your interest! I've enabled issues on the Warhammer repository and migrated your issue over here 😊 - I'll get back to you with some details on Warhammer API.

**Comment by brittonhayes** [comment]
Alrighty, I've got some time now and happy to help! 👋

To give you some details, as of right now the API isn't yet hosted anywhere on the web unfortunately. My domain [aos-api.com](https://aos-api.com) is running a legacy version of the Warhammer API from ~2 years ago that isn't as good as it could be. Eventually I'd like to host the shiny new API at that domain for anyone to use, but I haven't decided how I want to deploy it in a cost-effective manner yet.

With all that said, the API is 100% available to you in "early-access" to self-host by running it locally, you can run the API with [Docker](https://docs.docker.com/get-docker/) by following the quick start instructions:

```shell
# 1. Install docker 📥
https://docs.docker.com/get-docker/

# 2. Start containers 📦
docker-compose up --build --force-recreate --remove-orphans --detach

# 3. Try out the API ✨ 
http://warhammer.localhost:8090/api/grand-strategies
```

Example Response ->

```json
[
  {
    "_id": "hold_the_line",
    "description": "Hold the majority of the objectives at the end of the fifth battle round.",
    "name": "Hold the Line"
  },
  {
    "_id": "arcane_power",
    "description": "Successfully cast 3 spells with a casting value of 7 or higher during the battle.",
    "name": "Arcane Power"
  }
]
```

This API is currently in a state where I am looking for help from the community to add data. If there's a missing unit, allegiance, grand strategy etc., I welcome any and all pull requests to get that stuff added or corrected! 

I hope this is helpful and gives you some options @morteeke. I know it's not the most satisfying answer, but hopefully I can get the self-hosting simplified as much as possible so you can just run `docker run -p 8080:8080 ghcr.io/brittonhayes/warhammer:latest` and boom you'll have an API locally. (At least until I get it hosted)

**Reply by brittonhayes** [reply]
@morteeke Soooo slight update, I got the API hosted on the web for ya. Turned out to not be too difficult. No API key needed, have fun! 🎉 

Here's the URL for the API:
- API: https://aos-api.com/
- Documentation: https://aos-api.com/docs

**Reply by morteeke** [reply]
Many thanks! I will check it out.

I will help you with adding data where I can.
